ngraph/CMakeLists.txt

```diff
@@ -464,6 +463,9 @@ if (NGRAPH_ONNX_IMPORT_ENABLE)
         set(protobuf_WITH_ZLIB  OFF CACHE BOOL "" FORCE)
         set(protobuf_MSVC_STATIC_RUNTIME OFF CACHE BOOL "Link protobuf to static runtime libraries" FORCE)
     endif()
+
+    set(BEFORE_ONNX_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
```

Comment:
Can we use `set_target_properties`?

It should allow to avoid global properties.

Reply:
there is no such property to set shared or static library. cmake has `TYPE` property which is read-only https://cmake.org/cmake/help/v3.0/prop_tgt/TYPE.html?highlight=type

Reply:
Yeah hight, got it.